config/formats.js

```diff
@@ -226,6 +226,34 @@ exports.Formats = [
+			const megaStone = ["Abomasite", "Absolite", "Aerodactylite", "Aggronite", "Alakazite", "Altarianite", "Ampharosite", "Audinite", "Banettite", "Beedrillite", "Blastoisenite", "Blazikenite", "Cameruptite", "Charizardite X", "Charizardite Y", "Diancite", "Galladite", "Garchompite", "Gardevoirite", "Gengarite", "Glalitite", "Gyaradosite", "Heracronite", "Houndoominite", "Kangaskhanite", "Latiasite", "Latiosite", "Lopunnite", "Lucarionite", "Manecite", "Mawilite", "Medichamite", "Metagrossite", "Meteorite", "Mewtwonite X", "Mewtwonite Y", "Pidgeotite", "Pinsirite", "Sablenite", "Salamencite", "Sceptilite", "Scizorite", "Sharpedonite", "Slowbronite", "Steelixite", "Swampertite", "Tyranitarite", "Venusaurite"];
+			const zCrystal = ["Aloraichium Z", "Buginium Z", "Darkinium Z", "Decidium Z", "Dragonium Z", "Eevium Z", "Electrium Z", "Fairium Z", "Fightinium Z", "Firium Z", "Flyinium Z", "Ghostium Z", "Grassium Z", "Groundium Z", "Icium Z", "Incinium Z", "Kommonium Z", "Lunalium Z", "Lycanium Z", "Marshadium Z", "Mewnium Z", "Mimikium Z", "Normalium Z", "Pikanium Z", "Pikashunium Z", "Poisonium Z", "Primarium Z", "Psychium Z", "Rockium Z", "Snorlium Z", "Solganium Z", "Steelium Z", "Tapunium Z", "Ultranecrozium Z", "Waterium Z"];
+			const template = this.getTemplate(set.species || set.name);
+			if (!sinnohDex.includes(template.baseSpecies) && !sinnohDex.includes(template.species) && template.num > 251) {
+				return [`${set.name || set.species} is not in the Sinnoh Pokédex.`];
+			}
+			if (!megaStone.includes(template.item)) {
```

Comment:
You could just do something like ```
if (this.getItem(set.item).megaStone) {
    return [`Mega Stones are banned in Ultra Sinnoh Classic.`, `${set.name} is holding ${this.getItem(set.item).name}.`];
}```

And repeat something similar for z moves, eliminating the need for the megaStone and zCrystal consts